source/transactions/transactions.rst

```diff
@@ -226,14 +226,11 @@ readConcern to the first command in a transaction if and only if the
 readConcern is supplied and not the default. Drivers MUST NOT add to
 subsequent commands the readConcern from the transaction or any
 readConcern inherited from the collection, database, or client. If the
-user supplies an explicit readConcern via a method option, however,
-drivers MUST apply the readConcern, which will result in a server error.
-See `Users can pass prohibited options to operations in transactions`_.
+user supplies an explicit readConcern via a method option, the driver MUST
+raise an error with the message "Cannot set read concern after starting a
```

Comment:
In this case, should the transaction also be aborted? Suppose the user starts a transaction and executes the first command correctly (without specifying a read concern). But then a later command includes a read concern, causing a driver error. The transaction isn't aborted, and the user could still commit the transaction if they ignored the error. Is that okay? Would the transaction have been aborted before, when we passed read concern to the server?

Reply:
A quick test shows that with the current spec (before this PR) you could do:
```
start transaction with readConcern majority
do a find_one in the txn, successfully
do a find_one in the txn with readConcern snapshot
  server replies "Only the first command in a transaction may specify a readConcern"
do a find_one in the txn, successfully
```
The server error does *not* abort the transaction. So if we say that the new client-side error doesn't abort the transaction either then this is a smaller change. I also think it's consistent with a principle that applies to other scenarios: incorrectly specified commands cause parse errors that don't abort the transaction, whereas errors caused by operations on the data (like a write conflict or oversized transaction) do abort it.

Reply:
Ok, that makes sense.